app/services/appointment.js

```diff
@@ -0,0 +1,49 @@
+const CrudService = require('./crudService');
+const companyService = require('./company');
+const error = require('../services/error');
+
+class AppointmentService extends CrudService {
+  async create(appointment, companyId) {
+    if (appointment.id !== undefined) {
+      error.throwValidationError('Invalid appointment format.');
+    }
+
+    const appointmentExists = await this.repository.getByPropsNonParanoid({ ...appointment });
```

Comment:
I think here you should also validate that this appointment doesn't overlap with another one.